osu.Game/Screens/Play/HUD/HitErrorDisplay.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Rulesets.Judgements;
+using osuTK.Graphics;
+using osuTK;
+using osu.Framework.Graphics.Sprites;
+using System.Collections.Generic;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Beatmaps;
+using osu.Framework.Bindables;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Colour;
+using osu.Game.Graphics;
+using osu.Framework.Extensions.Color4Extensions;
+using System.Linq;
+using osu.Game.Configuration;
+
+namespace osu.Game.Screens.Play.HUD
+{
+    public class HitErrorDisplay : CompositeDrawable
+    {
+        private const int stored_judgements_amount = 5;
+        private const int bar_width = 3;
+        private const int judgement_line_width = 8;
+        private const int bar_height = 200;
+        private const int fade_duration = 200;
+        private const int arrow_move_duration = 500;
+        private const int judgement_life_time = 10000;
+        private const int spacing = 3;
+
+        public HitWindows HitWindows { get; set; }
+
+        [Resolved]
+        private Bindable<WorkingBeatmap> beatmap { get; set; }
+
+        [Resolved]
+        private OsuColour colours { get; set; }
+
+        private readonly SpriteIcon arrow;
+        private readonly FillFlowContainer bar;
+        private readonly Container judgementsContainer;
+        private readonly Queue<double> judgementOffsets = new Queue<double>();
+
+        private readonly Bindable<ScoreMeterType> type = new Bindable<ScoreMeterType>();
+
+        public HitErrorDisplay(bool reversed = false)
+        {
+            AutoSizeAxes = Axes.Both;
+
+            AddInternal(new FillFlowContainer
+            {
+                AutoSizeAxes = Axes.X,
+                Height = bar_height,
+                Direction = FillDirection.Horizontal,
+                Spacing = new Vector2(spacing, 0),
+                Children = new Drawable[]
+                {
+                    judgementsContainer = new Container
+                    {
+                        Anchor = reversed ? Anchor.CentreRight : Anchor.CentreLeft,
```

Comment:
can't this be done by rotating the entire display 180 degrees from the HUD?

Reply:
Think about it: When the left arrow will go down - the arrow of the second one will go up